snuba/subscriptions/data.py

```diff
@@ -0,0 +1,45 @@
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import Sequence
+
+from snuba.datasets.dataset import Dataset
+from snuba.query.query import Aggregation
+from snuba.query.types import Condition
+from snuba.request import Request
+from snuba.request.schema import RequestSchema
+from snuba.utils.metrics.timer import Timer
+from snuba.views import validate_request_content
+
+
+@dataclass(frozen=True)
+class Subscription:
+    """
+    Represents the state of a subscription.
+    """
+    id: str
+    project_id: int
+    conditions: Sequence[Condition]
+    aggregations: Sequence[Aggregation]
+    time_window: timedelta
+    resolution: timedelta
+
+    def build_request(self, dataset: Dataset, timestamp: datetime, offset: int, timer: Timer) -> Request:
```

Comment:
why isn't the dataset part of the subscription and instead passed at the moment the request is built ? I mean conceptually. Is it a serialization problem ? Specifically who is going to provide it  and how do you plan disambiguate multiple subscription designed for different datasets ?

Reply:
https://github.com/getsentry/snuba/pull/618#discussion_r354089337

I don't feel strongly either way about this.